solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/acknowledge/TransactedJCSMPAcknowledgementCallback.java

```diff
@@ -0,0 +1,98 @@
+package com.solace.spring.cloud.stream.binder.inbound.acknowledge;
+
+import com.solace.spring.cloud.stream.binder.util.ErrorQueueInfrastructure;
+import com.solace.spring.cloud.stream.binder.util.SolaceAcknowledgmentException;
+import com.solacesystems.jcsmp.JCSMPException;
+import com.solacesystems.jcsmp.transaction.RollbackException;
+import com.solacesystems.jcsmp.transaction.TransactedSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.integration.acks.AcknowledgmentCallback;
+
+class TransactedJCSMPAcknowledgementCallback implements AcknowledgmentCallback {
+	private final ThreadLocal<TransactedSession> transactedSessionThreadLocal = new ThreadLocal<>();
```

Comment:
Thread local should be static, if it were an instance level field, then it would actually be "Per Thread - Per Instance", not just a guaranteed "Per Thread."

Reply:
The main goal behind trying to use `ThreadLocal` was just to throw an exception if `acknowledge(Status)` was invoked from a different thread from the thread that created this instance.

Instead of using `ThreadLocal`, I can cache the thread ID and check that in the `acknowledge(Status)` method.

Reply:
that would be ideal, to not use ThreadLocal in this context

Reply:
Replaced usage of `ThreadLocal` to instead use a thread ID cached during object creation.